lib/excon/utils.rb

```diff
@@ -64,27 +64,21 @@ def split_header_value(str)
 
     # Escapes HTTP reserved and unwise characters in +str+
     def escape_uri(str)
-      str = str.dup
       str.force_encoding('BINARY') if FORCE_ENC
```

Comment:
It looks, that ```force_encoding``` [doesn't change internal byte representation of the string](http://stackoverflow.com/questions/21614228/what-is-the-difference-betwee-encode-and-force-encoding-in-ruby)
```gsub``` method doen't modify str, only returned modified string